tests/test_replace_im_links.py

```diff
@@ -373,3 +376,157 @@ def test_fetch_im_resource_extracted_no_matches(
     assert len(resources) == 2
     for r in resources:
         assert new_prefix not in r
+
+
+def test_replace_im_resources_html_overlapping_filenames(
+    tmp_path, mocker, requests_mock
+):
+    media_json = [
+        {
+            'mime_type': 'application/json',
+            'sha1': 'c527d2ba1d2dbcefb509822abb0bf2ab04457a2c',
+            'original_filename': 'abcd.json',
+            's3_key': 'resources/c527d2ba1d2dbcefb509822abb0bf2ab04457a2c'
+        }
+    ]
+    json_object = json.dumps(media_json, indent=4)
+    media_path = tmp_path / "media.json"
+    with open(media_path, "w") as outfile:
+        outfile.write(json_object)
+
+    filename = "abcd.json"
+    filename2 = "efgh.json"
+    im_resource = f"{IM_PREFIX}{filename}"
+    im_resource2 = f"{IM_PREFIX}{filename2}"
+
+    resource_content = b"123456789abcdef"
+    requests_mock.get(
+        im_resource2,
+        content=resource_content
+    )
+
+    lesson1_page1_content = f'''
+    <div><p>Lesson 1 Page 1</p>
+    <img src="{im_resource}"></img></div>
+    <img src="{im_resource2}"></img></div>'''
+
+    extracted_path = tmp_path / "extracted"
+    extracted_path.mkdir(parents=True, exist_ok=True)
+    extracted_filename = extracted_path / "lesson1page1.html"
+    extracted_filename.write_text(lesson1_page1_content)
+
+    new_prefix = "k12"
+    mocker.patch(
+        "sys.argv",
+        ["", f"{extracted_path}", f"{media_path}", f"{new_prefix}", "html"]
+    )
+    replace_im_links.main()
+
+    resources = []
+    for filename in os.listdir(extracted_path):
+        f = os.path.join(extracted_path, filename)
+        with open(f, 'r') as file:
+            data = file.read()
+            for elem in html.fragments_fromstring(
+                        data)[0].xpath('//*[@src]'):
+                resources.append(elem.attrib['src'])
+    assert len(resources) == 2
+    for r in resources:
+        assert new_prefix in r
+
+
+def test_fetch_and_replace_im_resources_html_overlapping_filenames(
```

Comment:
@mjoss17 - Ah, I see now what you meant by the test case you added. It actually doesn't seem too bad to me in terms of maintainability versus the value it provides in terms of providing some sanity checking that the three related commands "fit" together. Thank you for thinking of / adding this!